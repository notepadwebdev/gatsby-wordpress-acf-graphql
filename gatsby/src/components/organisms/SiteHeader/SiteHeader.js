import React from 'react';
import { useStaticQuery, graphql, Link } from 'gatsby';
import PrimaryNavigation from '../PrimaryNavigation/PrimaryNavigation';
import './SiteHeader.scss';

const SiteHeader = ({ site }) => {
  const data = useStaticQuery(graphql`
    query {
      wp {
        generalSettings {
          title
        }
      }
    }
  `);

  return (
    <header className="site-header">
      <div className="container">
        <h2>
          <Link to="/">{data.wp.generalSettings.title}</Link>
        </h2>
        <PrimaryNavigation />
      </div>
    </header>
  );
};

export default SiteHeader;
