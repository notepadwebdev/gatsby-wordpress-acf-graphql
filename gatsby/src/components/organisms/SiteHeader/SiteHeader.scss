.site-header {
  .container {
    display: flex;
    align-items: center;

    > * {
      &:first-child {
        flex-basis: 200px;
      }
    }
  }
}
