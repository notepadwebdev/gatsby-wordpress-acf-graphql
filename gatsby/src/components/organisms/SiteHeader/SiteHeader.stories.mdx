import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import SiteHeader from './SiteHeader';

<Meta
  title="Organisms/SiteHeader"
  component={SiteHeader}
/>

export const Template = (args) => <SiteHeader {...args} />;

export const Default = Template.bind({});
Default.args = {
  menuItems: {
    nodes: [
      {
        id: `bmF2X21lbnVfaXRlbTo4`,
        label: `Sample Page`,
        path: `/sample-page/`
      },
      {
        id: `bmF2X21lbnVfaXRlbToxNg==`,
        label: `Privacy Policy`,
        path: `/privacy-policy/`
      }
    ]
  }
}

# SiteHeader

The main header for the site.

<ArgsTable of={SiteHeader} />

## Stories
---

<Canvas>
  <Story name="Default" args={Default.args}>
    {Template.bind({})}
  </Story>
</Canvas>
