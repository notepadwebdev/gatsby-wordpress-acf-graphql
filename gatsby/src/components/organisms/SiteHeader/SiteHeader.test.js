import React from 'react';
import renderer from 'react-test-renderer';
import SiteHeader from './SiteHeader';

describe(`Site Header component`, () => {
  it(`renders correctly`, () => {
    const component = renderer.create(
      <SiteHeader menuItems={{
        nodes: [
          {
            id: `bmF2X21lbnVfaXRlbTo4`,
            label: `Sample Page`,
            path: `/sample-page/`
          },
          {
            id: `bmF2X21lbnVfaXRlbToxNg==`,
            label: `Privacy Policy`,
            path: `/privacy-policy/`
          }
        ]
      }} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
