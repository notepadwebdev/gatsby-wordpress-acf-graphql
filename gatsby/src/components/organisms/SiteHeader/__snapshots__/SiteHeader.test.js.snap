// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Header component renders correctly 1`] = `
<header
  className="site-header"
>
  <div
    className="container"
  >
    <h2>
      <a
        href="/"
      >
        LOGO
      </a>
    </h2>
    <nav
      className="primary-navigation"
      role="navigation"
    >
      <ul>
        <li>
          <a
            href="/sample-page/"
          >
            Sample Page
          </a>
        </li>
        <li>
          <a
            href="/privacy-policy/"
          >
            Privacy Policy
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
